routes/api/comments/index.js

```diff
@@ -1,147 +1,111 @@
-
-  const {body, author_id, asset_id, parent_id, status, username} = req.body;
+  const {
+    body,
+    asset_id,
+    parent_id,
+    author_id
+  } = req.body;
 
   Comment
-    .new(body, author_id, asset_id, parent_id, status, username)
+    .create({
+      body,
+      asset_id,
+      parent_id,
+      status: req.wordlist.matched ? 'rejected' : '',
+      author_id
+    })
     .then((comment) => {
-      res.status(200).send({'id': comment.id});
+
+      res.status(201).send(comment);
     })
-    .catch(error => {
-      next(error);
+    .catch((err) => {
+      next(err);
     });
 });
 
-router.post('/:comment_id', (req, res, next) => {
+router.get('/:comment_id', (req, res, next) => {
   Comment
     .findById(req.params.comment_id)
-    .then((comment) => {
-      comment.body = req.body.body;
-      comment.author_id = req.body.author_id;
-      comment.asset_id = req.body.asset_id;
-      comment.parent_id = req.body.parent_id;
-      comment.status = req.body.status;
-      return comment.save();
-    })
-    .then((comment) => {
-      res.status(200).send(comment);
+    .then(comment => {
+      if (!comment) {
+        res.status(404).end();
+        return;
+      }
+
+      res.status(200).json(comment);
     })
-    .catch(error => {
-      next(error);
+    .catch((err) => {
+      next(err);
     });
 });
 
-router.post('/:comment_id/status', (req, res, next) => {
-
-  Comment
-    .changeStatus(req.params.comment_id, req.body.status)
-    .then(comment => res.status(200).send(comment))
-    .catch(error => next(error));
-
-});
-
-router.post('/:comment_id/actions', (req, res, next) => {
+router.delete('/:comment_id', (req, res, next) => {
   Comment
-    .addAction(req.params.comment_id, req.body.user_id, req.body.action_type)
-    .then((action) => {
-      res.status(200).send(action);
+    .removeById(req.params.comment_id)
+    .then(() => {
+      res.status(204).end();
     })
-    .catch(error => {
-      next(error);
+    .catch((err) => {
+      next(err);
     });
 });
 
-//==============================================================================
-// Delete Routes
-//==============================================================================
+router.put('/:comment_id/status', (req, res, next) => {
+
+  const {
+    status
+  } = req.body;
 
-router.delete('/:comment_id', (req, res, next) => {
   Comment
-    .removeById(req.params.comment_id)
+    .changeStatus(req.params.comment_id, status)
     .then(() => {
-      res.status(201).send({});
+      res.status(204).end();
     })
-    .catch(error => {
-      next(error);
+    .catch((err) => {
+      next(err);
     });
 });
 
-router.delete('/:comment_id/actions', (req, res, next) => {
-  console.log(req.params);
+router.post('/:comment_id/actions', (req, res, next) => {
+
+  const {
+    user_id,
+    action_type
+  } = req.body;
+
   Comment
-    .removeAction(req.params.comment_id, req.body.user_id, req.body.action_type)
```

Comment:
So we're now deleting actions by referencing the specific action id rather than the user, item_id and action type? Sounds fair, though we'll need some updates not included in this PR to make action deletion work.